service/src/main/java/eu/europa/ec/fisheries/ers/fa/dao/FishingTripDao.java

```diff
@@ -155,8 +152,8 @@ public Integer getCountOfFishingTripsForMatchingFilterCriteria(FishingActivityQu
         log.debug("SQL:" + sqlToGetActivityList);
         Query typedQuery = em.createQuery(sqlToGetActivityList.toString());
         Query query = search.fillInValuesForTypedQuery(queryDto, typedQuery);
-        Long nrOfFa = (Long) query.getResultList().get(0);
-        return nrOfFa.intValue();
+        List<Long> summary = query.getResultList();

```

Comment:
Don't bring all these results from the DB into Java's memory just to count them. We need to try to find a better solution.